### PR TITLE
refactor/product/card

### PR DESCRIPTION
**Pull request by Hutavi**
refactor/product/card

### CODE REVIEW

File: Frontend/user/src/components/Category/BaseCategory.vue [hunk 1]
```diff
@@ -132,10 +132,7 @@ onMounted(() => {
 });
 
 const idDefine = (index: number, idx: number | undefined = undefined) => {
-  if (idx !== undefined) {
-    return `id-${index}-${idx}`;
-  }
-  return `id-${index}`;
+  return !idx ? `id-${index}-${idx}` : `id-${index}`;
```
Comment (tranduykhuong) [inline Frontend/user/src/components/Category/BaseCategory.vue]: It seems this logic is different from the old code

**Review by tranduykhuong — CHANGES_REQUESTED** [review]
Please check logic again

File: Frontend/user/src/components/Category/BaseCategory.vue (hunk 1 above)
Comment (Hutavi) [reply Frontend/user/src/components/Category/BaseCategory.vue]: This logic was merged at the meeting and demo example with Minh and this was merged into the dev branch by Minh
When I checkout to the "refactor/products/tags" branch from the dev branch, it was there before

Comment (Hutavi) [reply Frontend/user/src/components/Category/BaseCategory.vue]: I see the logic of these two functions are the same

**Pull request merged**